metrics/api/src/main/java/io/helidon/metrics/api/RegistryFactoryManager.java

```diff
@@ -83,6 +85,8 @@ private static RegistryFactoryProvider loadRegistryFactoryProvider() {
         return provider;
     }
 
+    private static final Semaphore SETTINGS_ACCESS = new Semaphore(1);
```

Comment:
Same `Semaphore`-to-`Lock` comment as above.  (Maybe you have good reasons to use `Semaphore` instead of `Lock`? It's not the idiom I'd expect for simple same-thread lock/unlock semantics.)

Reply:
Fixed